jobclient/python/cook/scheduler/client/jobs.py

```diff
@@ -359,15 +361,20 @@ def from_dict(cls, d: dict) -> 'Job':
         if 'instances' in d:
             d['instances'] = list(map(Instance.from_dict, d['instances']))
         if 'uris' in d:
-            d['uris'] = list(map(FetchableUri.from_dict, d['uris']))
-        if 'constraints' in d:
-            d['constraints'] = set(map(constraints.parse_from, d['constraints']))  # noqa E501
+            d['uris'] = [uri['value'] for uri in d['uris']]
+        # Assigning the Job's group works in the following manner:
+        # * If the key 'group' is defined, use that UUID.
+        # * Otherwise, if the key 'groups' is defined and has more than one
+        #   item, use the first element of that list.
         if 'group' in d:
             d['group'] = UUID(d['group'])
```

Comment:
We should be able to get rid of this check. If the job is in a group, the API will always return `groups` (the list) with a single group UUID in it.